pennylane/ops/qubit.py

```diff
@@ -2505,6 +2505,92 @@ def diagonalizing_gates(self):
         """
         return [QubitUnitary(self.eigendecomposition["eigvec"].conj().T, wires=list(self.wires))]
 
+# =============================================================================
+# Arithmetic
+# =============================================================================
+
+class Carry(Operation):
+    r"""Carry()
+    Apply a ``Carry`` operation on input wires
+    TODO add more information
+
+    **Details:**
+
+    * Number of wires: 4
+    * Number of parameters: 0
+    TODO consider gradient
+    * Gradient recipe: TODO
+
+    Args:
+        wires (Union[Wires or int]): the wires the operation acts on
```

Comment:
There are two ways we could think about this:
- Either we just let the user pass in `wires` (must be 4 of them)
- We have 4 separate arguments, e.g., input carry, output carry, input A, input B

I'm in favour of the first option, but as long as we make it clear which wire does what. This is similar to [CNOT](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.CNOT.html), where we have a single `wires` argument.
![image](https://user-images.githubusercontent.com/49409390/112628903-f4551a80-8e09-11eb-91d5-0c8fceb869c0.png)

However, in going for the first option, we'll have to be more careful when we make an adder between two registers, since this `Carry` operation will work on a combination of those registers.

Reply:
I also prefer the first option. I think it should be okay as long as we are clear about which wires do what.